airbyte-integrations/connectors/source-lever-hiring/source_lever_hiring/streams.py

```diff
@@ -22,74 +22,150 @@
+
+    def path(self, **kwargs) -> str:
+        return self.name
+
+    @property
+    @abstractmethod
+    def schema(self) -> BaseSchemaModel:
+        """Pydantic model that represents stream schema"""
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        """
-        TODO: Override this method to define a pagination strategy. If you will not be using pagination, no action is required - just return None.
-
-        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
-        to most other methods in this class to help you form headers, request bodies, query params, etc..
-
-        For example, if the API accepts a 'page' parameter to determine which page of the result to return, and a response from the API contains a
-        'page' number, then this method should probably return a dict {'page': response.json()['page'] + 1} to increment the page count by 1.
-        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
-
-        :param response: the most recent response from the API
-        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
-                If there are no more pages in the result, return None.
-        """
-        return None
-
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        """
-        TODO: Override this method to define any query parameters to be set. Remove this method if you don't need to define request params.
-        Usually contains common params e.g. pagination size etc.
-        """
-        return {}
+        response_data = response.json()
+        if response_data.get("hasNext"):
+            return {"offset": response_data["next"]}
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = {"limit": self.page_size}
+        params.update(self.stream_params)
+        if next_page_token:
+            params.update(next_page_token)
+        return params
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        """
-        TODO: Override this method to define how a response is parsed.
-        :return an iterable containing each record in the response
-        """
-        yield {}
+        yield from response.json()["data"]
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """Use Pydantic schema"""
+        return self.schema.schema()
 
 
 class IncrementalLeverHiringStream(LeverHiringStream, ABC):
+
+    state_checkpoint_interval = 100
+    cursor_field = "updatedAt"
+
+    def __init__(self, start_dt: int, **kwargs):
```

Comment:
Why is it not `start_date`?

Reply:
same comment, and we later call it `start_ts` - ideally we should have a single name for it unless it is transformed